src/staking-v3/logic/services/IDappStakingService.ts

```diff
@@ -0,0 +1,145 @@
+    contractAddress: string,
+    amount: number,
+    senderAddress: string,
+    successMessage: string
+  ): Promise<void>;
+
+  /**
+   * Invokes claim staker rewards call.
+   * @param senderAddress Address of the request sender.
+   * @param successMessage Message to be displayed on the call success.
+   */
+  claimStakerRewards(senderAddress: string, successMessage: string): Promise<void>;
+
+  /**
+   * Invokes calls to claim all rewards and to unstake tokens from unregistered contract.
+   * @param senderAddress Address of the request sender.
+   * @param contractAddress Address of the contract to be un staked from.
+   * @param successMessage Message to be displayed on the call success.
+   */
+  claimAllAndUnstakeFromUnregistered(
+    senderAddress: string,
+    contractAddress: string,
+    successMessage: string
+  ): Promise<void>;
+
+  /**
+   * Calculates staker rewards.
+   * @param senderAddress Staker address.
+   * @returns Staker rewards amount.
+   */
+  getStakerRewards(senderAddress: string): Promise<bigint>;
+
+  /**
+   * Calculates a dApp rewards
+   * @param contractAddress Contract address to calculate reward for.
+   * @returns Dapp rewards amount.
+   */
+  getDappRewards(contractAddress: string): Promise<bigint>;
+
+  /**
+   * Invokes claim dapp rewards call.
+   * @param contractAddress Address of the contract to be claimed from.
+   * @param senderAddress Address of the request sender.
+   * @param successMessage Message to be displayed on the call success.
+   */
+  claimDappRewards(
+    contractAddress: string,
+    senderAddress: string,
+    successMessage: string
+  ): Promise<void>;
+
+  /**
+   * Calculates staker bonus rewards.
+   * @param senderAddress Staker address.
+   * @returns Staker bonus rewards amount.
+   */
+  getBonusRewards(senderAddress: string): Promise<bigint>;
+
+  /**
+   * Invokes claim bonus rewards call.
+   * @param senderAddress Address of the request sender.
+   * @param successMessage Message to be displayed on the call success.
+   */
+  claimBonusRewards(senderAddress: string, successMessage: string): Promise<void>;
+
+  /**
+   * Invokes calls to claim staker bonus rewards calls.
+   * @param senderAddress Address of the request sender.
+   * @param successMessage Message to be displayed on the call success.
+   */
+  claimStakerAndBonusRewards(senderAddress: string, successMessage: string): Promise<void>;
+
+  /**
+   * Batches and invokes multiple calls
+   * 1. Claims staker, dApp and bonus rewards if available
+   * 2. Locks a given tokens amount
+   * 3. Stakes a given tokens amount
+   * @param senderAddress Staker address.
+   * @param amountToLock Tokens to lock
+   * @param stakeInfo A map containing contract addresses and amounts to stake.
+   * @param dappsToClaim List of dApp owned by sender to claim dApp rewards for.
+   * @param unstakeFromAddress Address of the contract to be unstaked from.
+   * @param unstakeAmount Amount of tokens to unstake.
```

Comment:
This is magic function which batches a lot of calls, depending what user selected on voting page:
- claim staker and bonus rewards, if any
- locks tokens,
- unstakes tokens in case of nomination transfer
- stakes tokens

I updated the method comment a bit